addons/radiodirectionfinder/functions/fnc_beaconCleanup.sqf

```diff
@@ -0,0 +1,35 @@
+#include "script_component.hpp"
+/*
+ * Author: PabstMirror
+ *
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call ace_radiodirectionfinder_fnc_beaconCleanup
+ *
+ * Public: No
+ */
+
+TRACE_1("beaconCleanup",_this);
+
+[{
+    private _removed = false;
+    private _list = missionNamespace getVariable [QGVAR(signalList), []];
+    _list = _list select {
+        _x params ["_obj"];
+        if ((_obj isEqualType []) || {alive _obj} || {!isNull _obj && {!(_obj isKindOf "ACE_transmitter_base")}}) then {
+            true
+        } else {
+            _removed = true;
```

Comment:
you could replace that with a count before != count after.
And get rid of the whole if inside the select.